### PR TITLE
Add JSDoc to config options

### PR DESCRIPTION
**Pull request by mohsen1**
Since the config file is now typed with TypeScript types we can add JSDoc to the config to make it a better DX.

I'm not sure how this is the right approach. There is a lot of copy-pasting from the documentation files. I wonder if it's worth building an automation pipeline for printing those documentations into the JSDoc. 

Before completing this work I would like to see what everyone thinks about this?

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 18.8s | 18.2s | -651ms |
| buildDurationCached | 3.8s | 3.9s | ⚠️ +32ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +5.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.847 | 3.851 | 0 |
| / avg req/sec | 649.81 | 649.24 | ⚠️ -0.57 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.938 | 1.989 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 1289.79 | 1257.16 | ⚠️ -32.63 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 27 kB | 27 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.9 kB | 70.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 19.1s | 19.5s | ⚠️ +309ms |
| buildDurationCached | 3.9s | 3.8s | -105ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +5.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.773 | 3.736 | -0.04 |
| / avg req/sec | 662.56 | 669.16 | +6.6 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.913 | 1.997 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1306.7 | 1251.57 | ⚠️ -55.13 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 27.2 kB | 27.2 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 71.1 kB | 71.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: be8c8671c937edde64f47df0eba9c920d11fc362

**Comment by mohsen1** [comment]
@timneutkens thoughts?

**Review by ijjk — COMMENTED** [review]
Hi, it sounds good to add JSDoc comments to the config here although these snippets are a bit verbose. Could we shorten these to keep them more minimal?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 18.1s | 18.3s | ⚠️ +185ms |
| buildDurationCached | 7.2s | 6.9s | -243ms |
| nodeModulesSize | 359 MB | 359 MB | ⚠️ +5.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.74 | 3.766 | ⚠️ +0.03 |
| / avg req/sec | 668.53 | 663.91 | ⚠️ -4.62 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.983 | 1.919 | -0.06 |
| /error-in-render avg req/sec | 1260.81 | 1302.54 | +41.73 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 27.3 kB | 27.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 71.2 kB | 71.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.94 kB | 4.94 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.3 kB | 14.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 22.5s | 22.3s | -196ms |
| buildDurationCached | 7.1s | 7s | -80ms |
| nodeModulesSize | 359 MB | 359 MB | ⚠️ +5.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.734 | 3.807 | ⚠️ +0.07 |
| / avg req/sec | 669.44 | 656.61 | ⚠️ -12.83 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.968 | 2.009 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1270.17 | 1244.45 | ⚠️ -25.72 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 27.4 kB | 27.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 71.3 kB | 71.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| image-HASH.js gzip | 4.97 kB | 4.97 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.21 kB | 2.21 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.3 kB | 14.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: f500b967d6e85afff83d0d8566e88de4a8c8a673

File: packages/next/server/config-shared.ts [hunk 1]
```diff
@@ -43,13 +48,122 @@ export interface TypeScriptConfig {
...
+  relay?: {
+    src: string
+    artifactDirectory?: string
+    language?: 'typescript' | 'flow'
+  }
+}
+
+/**
+ * Next configuration object
+ * @see [configuration documentations](https://nextjs.org/docs/api-reference/next.config.js/introduction)
```
Comment (styfle) [inline packages/next/server/config-shared.ts:134]:
```suggestion
 * @see [configuration documentation](https://nextjs.org/docs/api-reference/next.config.js/introduction)
```

File: packages/next/server/config-shared.ts [hunk 2]
```diff
@@ -43,13 +48,122 @@ export interface TypeScriptConfig {
   tsconfigPath?: string
 }
 
-export type NextConfig = { [key: string]: any } & {
+export interface WebpackConfigContext {
+  /** Next.js root directory */
+  dir: string
+  /** Indicates if the compilation will be done in development */
```
Comment (styfle) [inline packages/next/server/config-shared.ts:54]:
```suggestion
  /** It's `true` for `next dev` compilation and `false` for `next build` compilation */
```

File: packages/next/server/config-shared.ts [hunk 3]
```diff
@@ -43,13 +48,122 @@ export interface TypeScriptConfig {
...
+  dev: boolean
+  /** It's `true` for server-side compilation, and `false` for client-side compilation */
+  isServer: boolean
+  /**  The build id, used as a unique identifier between builds */
+  buildId: string
+  config: NextConfigComplete
```
Comment (styfle) [inline packages/next/server/config-shared.ts:60]:
```suggestion
  /** The next.config.js merged with default values */
  config: NextConfigComplete
```

File: packages/next/server/config-shared.ts [hunk 4]
```diff
@@ -43,13 +48,122 @@ export interface TypeScriptConfig {
...
+   * Headers allow you to set custom HTTP headers for an incoming request path.
+   *
+   * @see [Headers configuration documentations](https://nextjs.org/docs/api-reference/next.config.js/headers)
+   */
   headers?: () => Promise<Header[]>
+
+  /**
+   * Rewrites allow you to map an incoming request path to a different destination path.
+   *
+   * @see [Rewrites configuration documentations](https://nextjs.org/docs/api-reference/next.config.js/rewrites)
```
Comment (styfle) [inline packages/next/server/config-shared.ts:165]:
```suggestion
   * @see [Rewrites configuration documentation](https://nextjs.org/docs/api-reference/next.config.js/rewrites)
```

File: packages/next/server/config-shared.ts [hunk 5]
```diff
@@ -58,126 +172,215 @@ export type NextConfig = { [key: string]: any } & {
         fallback: Rewrite[]
       }
   >
+
+  /**
+   * Redirects allow you to redirect an incoming request path to a different destination path.
+   *
+   * @see [Redirects configuration documentations](https://nextjs.org/docs/api-reference/next.config.js/redirects)
```
Comment (styfle) [inline packages/next/server/config-shared.ts:179]:
```suggestion
   * @see [Redirects configuration documentation](https://nextjs.org/docs/api-reference/next.config.js/redirects)
```

File: packages/next/server/config-shared.ts [hunk 6]
```diff
@@ -58,126 +172,215 @@ export type NextConfig = { [key: string]: any } & {
...
+   */
   generateBuildId?: () => string | null | Promise<string | null>
+
+  /** @see [Disabling ETag Configuration](https://nextjs.org/docs/api-reference/next.config.js/disabling-etag-generation) */
   generateEtags?: boolean
+
+  /** @see [Including non-page files in the pages directory](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory) */
   pageExtensions?: string[]
+
+  /** @see [Compression documentations](https://nextjs.org/docs/api-reference/next.config.js/compression) */
```
Comment (styfle) [inline packages/next/server/config-shared.ts:250]:
```suggestion
  /** @see [Compression documentation](https://nextjs.org/docs/api-reference/next.config.js/compression) */
```

File: packages/next/server/config-shared.ts [hunk 7]
```diff
@@ -58,126 +172,215 @@ export type NextConfig = { [key: string]: any } & {
...
+   *
+   * @default 60
+   */
   staticPageGenerationTimeout?: number
+
+  /**
+   * Add `"crossorigin"` attribute to generated `<script>` elements generated by `<Head />` or `<NextScript />` components
+   *
+   *
+   * @see [`crossorigin` attribute documentations](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin)
```
Comment (styfle) [inline packages/next/server/config-shared.ts:369]:
```suggestion
   * @see [`crossorigin` attribute documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin)
```

File: packages/next/server/config-shared.ts [hunk 8]
```diff
@@ -58,126 +172,215 @@ export type NextConfig = { [key: string]: any } & {
...
-    | null
+  /**
+   * Before continuing to add custom webpack configuration to your application make sure Next.js doesn't already support your use-case
+   *
+   * @see [Custom Webpack Config documentation](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config)
+   */
+  webpack?: NextJsWebpackConfig | null
 
+  /**
+   * By default Next.js will redirect urls with trailing slashes to their counterpart without a trailing slash.
```
Comment (styfle) [inline packages/next/server/config-shared.ts:202]: This description should explain the difference between `true` and `false` and then list what the default value is.

**Review by mohsen1 — COMMENTED** [review]
Address typo and accept suggestions

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 14.8s | 15.4s | ⚠️ +585ms |
| buildDurationCached | 5.7s | 6s | ⚠️ +301ms |
| nodeModulesSize | 359 MB | 359 MB | ⚠️ +9.07 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.018 | 2.915 | -0.1 |
| / avg req/sec | 828.3 | 857.54 | +29.24 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.256 | 1.374 | ⚠️ +0.12 |
| /error-in-render avg req/sec | 1990.91 | 1819.11 | ⚠️ -171.8 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 27.4 kB | 27.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 71.2 kB | 71.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.94 kB | 4.94 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.3 kB | 14.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 526 B | 526 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| buildDuration | 17.9s | 18.3s | ⚠️ +315ms |
| buildDurationCached | 5.7s | 5.7s | ⚠️ +53ms |
| nodeModulesSize | 359 MB | 359 MB | ⚠️ +9.07 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.197 | 3.138 | -0.06 |
| / avg req/sec | 782.04 | 796.56 | +14.52 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.337 | 1.263 | -0.07 |
| /error-in-render avg req/sec | 1869.39 | 1979.09 | +109.7 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 27.4 kB | 27.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 71.3 kB | 71.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| image-HASH.js gzip | 4.97 kB | 4.97 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.21 kB | 2.21 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.3 kB | 14.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mohsen1/next.js add-jsdoc-to-config | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: 9d8fbbf932a87168644d14dfcb858a080c41b93a

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 9d8fbbf932a87168644d14dfcb858a080c41b93a

`yarn testheadless test/integration/middleware/core/test/index.test.js` 
- Middleware base tests > dev mode > should override with rewrite externally correctly

<details>
<summary>Expand output</summary>

  ● Middleware base tests › dev mode › should override with rewrite externally correctly

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 404

      196 |     )
      197 |
    > 198 |     expect(res.status).toBe(200)
          |                        ^
      199 |     expect(await res.text()).toContain('Example Domain')
      200 |
      201 |     const browser = await webdriver(context.appPort, `${locale}/rewrites`)

      at Object.<anonymous> (integration/middleware/core/test/index.test.js:198:24)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by ijjk — APPROVED** [review]
Thanks for the PR!

**Pull request merged**